### PR TITLE
[feat] Updating HMAC KDF from 384 to 512 for ROM

### PR DESCRIPTION
**Pull request by mhatrevi**

### CODE REVIEW

File: rom/dev/src/flow/cold_reset/fw_processor.rs [hunk 1]
```diff
@@ -71,8 +71,8 @@ impl FirmwareProcessor {
             // SHA2-512/384 Accelerator
             sha2_512_384_acc: &mut env.sha2_512_384_acc,
 
-            // Hmac384 Engine
-            hmac384: &mut env.hmac384,
+            // Hmac Engine
+            hmac384: &mut env.hmac,
```
Comment (swenson) [inline rom/dev/src/flow/cold_reset/fw_processor.rs:75]: Rename to `hmac384`?

File: rom/dev/src/main.rs [hunk 2]
```diff
@@ -139,8 +139,8 @@ pub extern "C" fn rom_entry() -> ! {
             // SHA2-512/384 Accelerator
             sha2_512_384_acc: &mut env.sha2_512_384_acc,
 
-            // Hmac384 Engine
-            hmac384: &mut env.hmac384,
+            // Hmac Engine
+            hmac384: &mut env.hmac,
```
Comment (swenson) [inline rom/dev/src/main.rs:143]: Same here

File: sw-emulator/lib/periph/src/hmac.rs [hunk 3]
```diff
@@ -635,6 +641,12 @@ impl HmacSha {
     fn tag_write_complete(&mut self) {
         let key_id = self.tag_write_ctrl.reg.read(TagWriteControl::KEY_ID);
 
+        // let temp = &self.tag.as_bytes()[..self.key_len() * 4];
+        // println!("EMU: HMAC Tag: ");
+        // for i in 0..temp.len() {
+        //     print!("{:x}, ", temp[i]);
+        // }
+        // println!();
```
Comment (swenson) [inline sw-emulator/lib/periph/src/hmac.rs:649]: Leftover from debugging?
```suggestion
```

File: test/tests/caliptra_integration_tests/smoke_testdata/idevid_csr.der
Comment (swenson) [inline test/tests/caliptra_integration_tests/smoke_testdata/idevid_csr.der:1]: Why did this and the cert decrease in size by 1 byte?

File: drivers/test-fw/src/bin/hmac_tests.rs [hunk 4]
```diff
@@ -1015,9 +1017,9 @@ test_suite! {
     test_hmac4,
     test_hmac_kv_multiblock,
     test_hmac5,
-    test_kdf0,
-    test_kdf1,
-    test_kdf2,
+    test_kdf0_hmac384,
```
Comment (swenson) [inline drivers/test-fw/src/bin/hmac_tests.rs:1020]: Do we need to add tests for hmac512?

File: drivers/src/hmac384_kdf.rs
Comment (bluegate010) [inline drivers/src/hmac384_kdf.rs:1]: We should probably rename this file.

File: runtime/src/dpe_crypto.rs [hunk 5]
```diff
@@ -53,7 +53,7 @@ impl<'a> DpeCrypto<'a> {
             sha384,
             trng,
             ecc384,
-            hmac384,
+            hmac: hmac384,
```
Comment (bluegate010) [inline runtime/src/dpe_crypto.rs:56]: Just above we can rename `hmac384: &'a mut Hmac` to just `hmac: ...`

File: sw-emulator/lib/periph/src/hmac.rs (hunk 3 above)
Comment (mhatrevi) [reply sw-emulator/lib/periph/src/hmac.rs:649]: Removed.

File: test/tests/caliptra_integration_tests/smoke_testdata/idevid_csr.der
Comment (mhatrevi) [reply test/tests/caliptra_integration_tests/smoke_testdata/idevid_csr.der:1]: Looks like the encoded signature went from 103 to 102 bytes.

File: rom/dev/src/flow/cold_reset/fw_processor.rs (hunk 1 above)
Comment (mhatrevi) [reply rom/dev/src/flow/cold_reset/fw_processor.rs:75]: Updated.

File: rom/dev/src/main.rs (hunk 2 above)
Comment (mhatrevi) [reply rom/dev/src/main.rs:143]: Updated.

File: drivers/test-fw/src/bin/hmac_tests.rs (hunk 4 above)
Comment (mhatrevi) [reply drivers/test-fw/src/bin/hmac_tests.rs:1020]: Yes; We'll add that as a separate change though.

File: runtime/src/dpe_crypto.rs (hunk 5 above)
Comment (mhatrevi) [reply runtime/src/dpe_crypto.rs:56]: Updated.

File: drivers/src/hmac384_kdf.rs
Comment (mhatrevi) [reply drivers/src/hmac384_kdf.rs:1]: Renamed.

**Review by swenson — DISMISSED** [review]
LGTM

**Review by bluegate010 — APPROVED** [review]

**Review by swenson — APPROVED** [review]

**Pull request merged**